### ISSUE TITLE
readme suggests nonexisting lint 

### ISSUE THREAD

**Issue by matthiaskrgr** [rust-lang/rust-clippy#3210]
`clippy 0.0.212 (125907a 2018-09-17)`

The readme states
"
For this to work you have to use Clippy on the nightly toolchain: `cargo +nightly clippy`. If you 
want to use Clippy with the stable toolchain, you can stick to the old unscoped method to 
enable/disable Clippy lints until `tool_lints` are stable:
```rust
#![cfg_attr(feature = "cargo-clippy", allow(clippy_lint))]
```
"
However there is no `clippy_lint`lint that could be suppressed/allowed.

````
warning: unknown lint: `clippy_lint`
  --> src/lib.rs:16:45
   |
16 | #![cfg_attr(feature = "cargo-clippy", allow(clippy_lint))]
   |                                             ^^^^^^^^^^^
   |
   = note: #[warn(unknown_lints)] on by default
````

EDIT: repro sample:

````rust
#![cfg_attr(feature = "cargo-clippy", allow(clippy_lint))]
#![cfg_attr(feature = "cargo-clippy", warn(if_not_else))]
fn main() {
    println!("Hello, world!");
}
````

cc @flip1995

**Comment by flip1995** [comment]
Oh yeah `clippy_lint` does not exist, I just used it as a placeholder. Maybe I should have made this more obvious.

**Comment by matthiaskrgr** [comment]
Ah, so there actually is no way to work around this besides
`#![cfg_attr(feature = "cargo-clippy", allow(renamed_and_removed_lints))]` ?

**Comment by flip1995** [comment]
If you use Clippy on the stable or beta toolchain (`cargo +stable clippy`/`cargo +beta clippy`) the `tool_lints` warning won't be emitted (rust-lang/rust#54358) until they are stable. If you use Clippy on the nightly toolchain (`cargo +nightly clippy`), you should move to `tool_lints` with the
```rust
#![cfg_attr(feature = "cargo-clippy", feature(tool_lints)]
```
workaround.

The only time `#![cfg_attr(feature = "cargo-clippy", allow(renamed_and_removed_lints))]` should be used IMO is when you use clippy on both nightly and stable/beta toolchain. But I don't see why someone would want to do that.

**Comment by flip1995** [comment]
Yeah just `allow(renamed_and_removed_lints)` _really_ shouldn't be used. Than the cfg_attr variant might be good. We should stabilize the `tool_lints`. I try to do this until 1.31 stable.

**Comment by phansch** [comment]
tool_lints are stable now, so closing :lock:
